### PR TITLE
Explain `walkdir` docstring second example. 

### PR DESCRIPTION
**Pull request by nathanrboyer**
This is a follow-up documentation PR to #55476. I believe the second example in the `walkdir` docstring is still unintuitive since the result changes each time. I attempted a simple explanation, but I don't really know what I'm talking about. Hopefully someone else can explain what is happening better. Some additional discussion in [this Discourse post](https://discourse.julialang.org/t/find-all-files-named-findthis-csv-in-nested-subfolders-of-rootfolder/118096).

### CODE REVIEW

**Comment by LilithHafner** [comment]
`walkdir` returns a stateful iterator (much like `Iterators.Stateful`). The relevant paragraph from the Stateful docstring that explains this best to me is

```
  `Stateful` provides the regular iterator interface. Like other mutable
  iterators (e.g. `Base.Channel`), if iteration is stopped early (e.g. by a
  `break` in a `for` loop), iteration can be resumed from the same spot by
  continuing to iterate over the same iterator object (in contrast, an
  immutable iterator would restart from the beginning).
```

File: base/file.jl [hunk 1]
```diff
@@ -1102,6 +1102,7 @@ The directory tree can be traversed top-down or bottom-up.
 If `walkdir` or `stat` encounters a `IOError` it will rethrow the error by default.
 A custom error handling function can be provided through `onerror` keyword argument.
 `onerror` is called with a `IOError` as argument.
+The returned iterator is mutable, so repeatedly accessing it will change its state.
```
Comment (LilithHafner) [inline base/file.jl:1105]:
```suggestion
The returned iterator is stateful so when accessed repeatedly each access will
resume where the last left off, like [`Iterators.Stateful`](@ref).
```

**Review by LilithHafner — APPROVED** [review]
LGTM! Does this make sense & seem ready to merge to you, too?

**Comment by nathanrboyer** [comment]
Yep! It gives the user an expectation for the behavior and a link to dig deeper if desired. Thanks!

**Comment by nathanrboyer** [comment]
As a funny aside, I spent the last two PRs thinking "LGTM" meant "Let's get to merge" and didn't bother looking it up until just now, but that did result in a pretty fast turnaround for these PRs 😆

**Pull request merged**